src/main/java/org/opentripplanner/model/plan/StopArrival.java

```diff
@@ -1,55 +1,31 @@
 package org.opentripplanner.model.plan;
 
-import java.time.ZonedDateTime;
 import org.opentripplanner.framework.tostring.ToStringBuilder;
 
 /**
  * This class is used to represent a stop arrival event mostly for intermediate visits to a stops
  * along a route.
+ *
+ * @param arrival          The time the rider will arrive at the place.
+ * @param departure        The time the rider will depart the place.
+ * @param stopPosInPattern For transit trips, the stop index (numbered from zero from the start of
+ *                         the trip).
+ * @param gtfsStopSequence For transit trips, the sequence number of the stop. Per GTFS, these
+ *                         numbers are increasing.
  */
-public class StopArrival {
-
-  public final Place place;
-  /**
-   * The time the rider will arrive at the place.
-   */
-  public final ZonedDateTime arrival;
-
-  /**
-   * The time the rider will depart the place.
-   */
-  public final ZonedDateTime departure;
-
-  /**
-   * For transit trips, the stop index (numbered from zero from the start of the trip).
-   */
-  public final Integer stopPosInPattern;
-
-  /**
-   * For transit trips, the sequence number of the stop. Per GTFS, these numbers are increasing.
-   */
-  public final Integer gtfsStopSequence;
-
-  public StopArrival(
-    Place place,
-    ZonedDateTime arrival,
-    ZonedDateTime departure,
-    Integer stopPosInPattern,
-    Integer gtfsStopSequence
-  ) {
-    this.place = place;
-    this.arrival = arrival;
-    this.departure = departure;
-    this.stopPosInPattern = stopPosInPattern;
-    this.gtfsStopSequence = gtfsStopSequence;
-  }
-
+public record StopArrival(
```

Comment:
records should not be used in the internal domain model. The are ment for DTOs, not rich domain objects.

Reply:
Reverted in https://github.com/opentripplanner/OpenTripPlanner/pull/5660/commits/f723c25a421b00c58fb7b1e32ca29ee31583e41c